packages/core/src/types/asyncapi.ts

```diff
@@ -0,0 +1,1141 @@
+import { NodeType, listOf, mapOf } from '.';
+import { isMappingRef } from '../ref-utils';
+
+const Root: NodeType = {
+  properties: {
+    asyncapi: null, // TODO validate semver format and supported version
+    info: 'Info',
+    id: { type: 'string' },
+    servers: 'ServerMap',
+    channels: 'ChannelMap',
+    components: 'Components',
+    tags: 'TagList',
+    externalDocs: 'ExternalDocs',
+    defaultContentType: { type: 'string' },
+  },
+  required: ['asyncapi', 'channels', 'info'],
+};
+
+const Channel: NodeType = {
+  properties: {
+    description: { type: 'string' },
+    subscribe: 'Operation',
+    publish: 'Operation',
+    parameters: 'ParametersMap',
+    bindings: 'ChannelBindings',
+    servers: { type: 'array', items: { type: 'string' } },
+  },
+};
+
+const ChannelMap: NodeType = {
+  properties: {},
+  additionalProperties: 'Channel',
+};
+
+const ChannelBindings: NodeType = {
+  properties: {},
+  allowed() {
+    // allow all supported values, not all have deep linting
```

Comment:
Looks like a TODO comment. If it is, please use the `TODO:` notation.

Reply:
There are several comments like this.

Reply:
It's not a TODO, it's an explanation. We support all the possible protocol values here, but not all of them have linting for that protocol. This is so that we won't produce errors if we get an API description that we can't verify in detail, but we aren't guaranteeing detailed support for them all.

Does this make sense, and do we need to do anything else?

Reply:
Oh, thanks for the explanation! I just wanted to be sure that we're using TODO:s consistently.